data-raw/YBFMP.R

```diff
@@ -12,55 +13,194 @@ Mode <- function(x) {
+    Date = mdy(SampleDate, tz = "America/Los_Angeles"),
+    Datetime = mdy_hm(Datetime, tz = "America/Los_Angeles"),
+    Station = StationCode,
+    Tide = recode(Tide, High="High Slack", Low="Low Slack"),
+    Temperature = WaterTemp,
+    # Convert Secchi to cm
+    Secchi = Secchi * 100,
+    # Convert Conductivity to Specific Conductance, see suisun.R for info
+    Conductivity = if_else(
+      is.na(SpecificConductance),
+      Conductivity / (1 + 0.019 * (Temperature - 25)),
+      SpecificConductance
+    ),
+    Microcystis = MicrocystisRank,
+    Notes = FieldComments
+  ) %>%
+  # Remove records where all parameters are NA
+  filter(!if_all(where(is.numeric), is.na)) %>%
+  # Remove duplicated records across all columns
+  distinct()
+
+# Pull out duplicated records with identical Station-Datetime combinations from fish data
+YBFMP_fish_dups <- YBFMP_fish %>%
+  count(Station, Datetime) %>%
+  filter(n > 1) %>%
+  select(-n) %>%
+  left_join(YBFMP_fish)
+
+# Clean up the duplicated records
+YBFMP_fish_dups_c <- YBFMP_fish_dups %>%
+  # Most of the duplicated records are due to different values in the Notes column
+  distinct(Date, Datetime, Station, Tide, Temperature, Secchi, Conductivity, Microcystis, .keep_all = TRUE) %>%
+  # Remove the one record with an NA value for Secchi since its pair has a value
+  drop_na(Secchi)
+
+# Add the cleaned up duplicate data to the original fish data set
+YBFMP_fish_c <- YBFMP_fish %>%
+  anti_join(YBFMP_fish_dups) %>%
+  bind_rows(YBFMP_fish_dups_c) %>%
+  # There are numerous records that were collected on the same day and station
+  # that have identical water quality measurements. After speaking with Nicole
+  # Kwan (SES Supervisor for the AES Unit), we decided to only keep the records
+  # with the earliest Datetime for the groups of records that share identical
+  # water quality measurements with different Datetimes.
+  group_by(Date, Station, Temperature, Secchi, Conductivity, Microcystis) %>%
+  filter(Datetime == min(Datetime)) %>%
+  ungroup() %>%
+  # There are two records (STTD on 2017-01-23) that share identical Temperature,
+  # Secchi, and Conductivity, but only one record has a Microcystis value (the
+  # record with the later Datetime). For this instance, we'll keep both records
+  # and convert Temperature, Secchi, and Conductivity to NA for the record
+  # with a reported Microcystis value.
+  mutate(
+    Temperature = if_else(Station == "STTD" & Datetime == "2017-01-23 13:30:00", NA_real_, Temperature),
+    Secchi = if_else(Station == "STTD" & Datetime == "2017-01-23 13:30:00", NA_real_, Secchi),
+    Conductivity = if_else(Station == "STTD" & Datetime == "2017-01-23 13:30:00", NA_real_, Conductivity)
+  )
+
+# Combine the zoop and fish WQ data sets
+YBFMP_comb <- bind_rows(YBFMP_zoop, YBFMP_fish_c) %>%
+  # Remove the duplicated records shared between the two data sets (STTD)
+  distinct()
+
+# Pull out the remaining duplicated records shared between the two data sets
```

Comment:
Any idea why they would have duplicates among the two data sets? I thought fish and zoop data were collected separately?

Reply:
I also asked Nicole about this as well, and she suspected that there were a few instances at STTD when the WQ measurements for the fish and zoop data were shared and duplicated into both databases. We decided to only keep one of the records.

Reply:
Great, thanks!